packages/language-server/src/codeActionProvider.ts

```diff
@@ -0,0 +1,189 @@
+  let bestCandidate: string | undefined
+  let justCheckExactMatches = false
+  const nameLowerCase = name.toLowerCase()
+  for (const candidate of relations) {
+    if (
+      !(
+        Math.abs(candidate.length - nameLowerCase.length) <=
+        maximumLengthDifference
+      )
+    ) {
+      continue
+    }
+    const candidateNameLowerCase = candidate.toLowerCase()
+    if (candidateNameLowerCase === nameLowerCase) {
+      return candidate
+    }
+    if (justCheckExactMatches) {
+      continue
+    }
+    if (candidate.length < 3) {
+      // Don't bother, user would have noticed a 2-character name having an extra character
+      continue
+    }
+    // Only care about a result better than the best so far.
+    const distance = levenshtein(nameLowerCase, candidateNameLowerCase)
+    if (distance > bestDistance) {
+      continue
+    }
+    if (distance < 3) {
+      justCheckExactMatches = true
+      bestCandidate = candidate
+    } else {
+      bestDistance = distance
+      bestCandidate = candidate
+    }
+  }
+  return bestCandidate
+}
+
+export function quickFix(
+  textDocument: TextDocument,
+  params: CodeActionParams,
+): CodeAction[] {
+  const lines: string[] = convertDocumentTextToTrimmedLineArray(textDocument)
+  const diagnostics: Diagnostic[] = params.context.diagnostics
+
+  if (isNullOrUndefined(diagnostics) || diagnostics.length === 0) {
+    return []
+  }
+
+  const codeActions: CodeAction[] = []
+
+  diagnostics.forEach((diag) => {
+    if (
+      diag.severity === DiagnosticSeverity.Error &&
+      diag.message.startsWith('Type') &&
+      diag.message.includes(
+        'is neither a built-in type, nor refers to another model, custom type, or enum.',
+      )
+    ) {
+      const diagText = textDocument.getText(diag.range).replace('?', '').replace('[]', '')
+      const spellingSuggestion = getSpellingSuggestionForModelAndEnumName(
+        diagText,
+        getAllRelationNames(lines),
+      )
+      if (spellingSuggestion) {
+        codeActions.push({
+          title: "Change spelling to '" + spellingSuggestion + "'",
+          kind: CodeActionKind.QuickFix,
+          diagnostics: [diag],
+          edit: {
+            changes: {
+              [params.textDocument.uri]: [
+                {
+                  range: diag.range,
+                  newText: spellingSuggestion,
+                },
+              ],
+            },
+          },
+        })
+      }
+      codeActions.push({
+        title: 'Make this relation and create model',
+        kind: CodeActionKind.QuickFix,
+        diagnostics: [diag],
+        edit: {
+          changes: {
+            [params.textDocument.uri]: [
+              {
+                range: insertModelOrEnumAtRange(textDocument),
+                newText: '\nmodel ' + diagText + ' {\n\n}\n',
+              },
+            ],
+          },
+        },
+      })
+      codeActions.push({
+        title: 'Make this relation and create enum',
```

Comment:
Technically an enum might not always be a "relation" but it can also be a "scalar". This wording might need to be adapted.